docs/user/kudu/index_config.rst

```diff
@@ -0,0 +1,145 @@
+Kudu Index Configuration
+========================
+
+GeoMesa exposes a variety of configuration options that can be used to customize and optimize a given installation.
+This section contains Kudu-specific options; general options can be found under :ref:`index_config`.
+
+Default Index Creation
+----------------------
+
+By default GeoMesa will only create a single Z3 (or XZ3) index for Kudu. Due to predicate push-downs,
+partition pruning and other advanced optimizations, queries on non-indexed fields are still quite fast.
+To create additional indices, specify them explicitly. See :ref:`customizing_index_creation` for details.
+
+Note that because they do not have an incrementing time field in the primary key, other indices may eventually
+run out of space. Initial partitioning is especially important in this case (see next).
+
+Table Partitioning
+------------------
+
+Partitioning is import in Kudu, and GeoMesa supports both Kudu hash and range partitions. See
+:ref:`configuring_z_shards` and :ref:`configuring_attr_shards` for details on configuring shards, which will be
+translated to Kudu hash partitions. See :ref:`table_split_config` for details on configuring table splits,
+which will be translated to Kudu range partitions. By default, the Z3/XZ3 index will create a new range partition
+for every time period (week by default).
+
+See the `Kudu documentation <http://kudu.apache.org/docs/schema_design.html#partitioning>`__ for more details on Kudu
+partitioning.
+
+Column Encoding
+---------------
+
+Kudu has multiple ways to encode data. Each column (attribute) in a schema is encoded separately. The best
+encoding will depend on the data being written; for example low-cardinality string columns work will with
+dictionary encoding. If nothing is specified, then the Kudu default encoding will be used based on the attribute
+type.
+
+Valid encodings depend on the attribute type:
+
+============================== ============================== ===========
+Attribute Type                 Encodings                      Default
+============================== ============================== ===========
+Integer, Long, Date            plain, bit shuffle, run length bit shuffle
+Float, Double                  plain, bit shuffle             bit shuffle
+Boolean                        plain, run length              run length
+String, Bytes, UUID, List, Map plain, prefix, dictionary      dictionary
+Point                          plain, bit shuffle             bit shuffle
+Non-point geometry             plain, prefix, dictionary      dictionary
```

Comment:
How does a dictionary encoding for non-point geometries work?

Reply:
same as for anything else - it's only useful if you have lots of repeated geometries. Kudu will transparently fall back to plain encoding if there are too many distinct values, so there shouldn't be much downside. Prefix seems useless, and plain doesn't do any encoding. Non-point geometries are stored as binary kryo, and dictionary is the default encoding for binary columns so I just left it.